docs/whiteblock-test-plan.md

```diff
@@ -0,0 +1,141 @@
+# RChain Test Plan
+
+## Overview
+
+The network has N validators plus one bootstrap node.
+
+## Test Procedure
+
+Per test case:
+1. Build network
+2. Provision nodes
+3. Configure network conditions between nodes according to specified test case
+4. Configure actions and behavior between nodes according to specified test case
+5. Output performance data in CSV format
+6. Aggregate data, parse, & present data
+8. Push data to appropriate repo
+9. Reset environment
+
+## Test Environment
+
+The proposed testing initiatives will be conducted within the Whiteblock test
+lab on a group of 6 rack mounted chassis servers.
+
+Should testing initiatives require additional computational resources or
+demand a higher overhead than can be adequately provided within the proposed
+environment, the Whiteblock cloud-based solution can be used. However, in the
+interest of cost-effectiveness, the current on-premises lab setup can
+adequately provide the current scope of work. 
+
+## System Specifications
+
+| Component   | Value                                          |
+|-------------|------------------------------------------------|
+| Model Name  | PowerEdge R720                                 |
+| Vendor      | Dell                                           |
+| CPUs        | (2) Intel Xeon CPU E5-2667 v2 @ 3.30GHz        |
+| CPU Max MHz | 4000.0000                                      |
+| NIC         | 82599ES 10-Gigabit SFI/SFP+ Network Connection |
+| RAM         | 256GB DDR4                                     |
+
+
+## Performance Metrics
+
+Time measurements are expressed in terms of the time passed on the node
+coordinating the tests.  Assuming the coordinating node's clock hasn't been tempered with.
+
+| Value			    | Description | 
+| ------------------------- | -------- | 
+| Discovery Time	    | The length of time from a node startup to node become aware of its first peer | 
+| Block Propagation Time    | The length of time it takes for a message, once broadcast, to be received by a majority (99%) of peers within the network |
+| Consensus Time	    | The length of time counted from a finish of a propose operation the all the network node's reporting the same tip hash | 
+| Node Join Time	    | The length of time from a node startup to adding first block to its DAG | 
+
+## Performance Tests
+
+The following tables define each test series within this test phase. A test
+series focuses on observing and documenting the effects of certain conditions
+on performance. Each test series is comprised of three separate test cases
+which define the variable to be tested. 
+
+### Series 1: Number of Validators
+
+| Variable         | Test Case A | Test Case B | Test Case C |
+|------------------|------------:|------------:|------------:|
+| Validators       | 300         | 600         | 1000        |
+| Static Nodes     | 600         | 600         | 600         |
+| Message Size     | 100KB       | 100KB       | 100KB       |
```

Comment:
Instead of `Message Size`, I think we need to settle on a contract to be `propose`d here.